### PR TITLE
Contracts & Harnesses for [f16, f128] to_int_unchecked

### PR DESCRIPTION
**Pull request by MWDZ**
Towards #59 

### Changes
- Added contracts for `f{16,128}::to_int_unchecked` (located in `library/core/src/num/f[16, 32].rs`)
- Added a macro for generating `to_int_unchecked` harnesses
- Added harnesses for `to_int_unchecked` of each integer type i8, i16, i32, i64, i128, isize, u8, u16, u32, u64, u128, usize --- 12 harnesses in total.
### Revalidation
1. Per the discussion in https://github.com/model-checking/verify-rust-std/issues/59, we have to build and run Kani from feature/verify-rust-std branch.
2. To revalidate the verification results, run the following command. <harness_to_run> can be either num::verify to run all harnesses or num::verify::<harness_name> (e.g. checked_f32_to_int_unchecked_i8) to run a specific harness.
```
kani verify-std  "path/to/library" \
    --harness <harness_to_run> \
    -Z unstable-options \
    -Z function-contracts \
    -Z mem-predicates
```
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Comment by Yenyun035** [comment]
Note: This PR is similar to PR #134 -- it depends on Kani Issue #3711. As far as I know, float_to_int_unchecked has not supported f16 and f128 (please correct me if this is resolved @celinval).

**Comment by Yenyun035** [comment]
> Note: This PR is similar to PR #134 -- it depends on Kani Issue [#3711](https://github.com/model-checking/kani/issues/3711).

Resolved. Once tested, it's ready for review.

**Comment by Yenyun035** [comment]
@celinval @zhassan-aws Why did f16 and f128 harnesses pass all Kani verification checks given that `float_to_int_unchecked` is not currently supported for f16 and f128?

**Comment by zhassan-aws** [comment]
@Yenyun035 Support for f16 and f128 was added in https://github.com/model-checking/kani/pull/3701.

**Comment by Yenyun035** [comment]
Depends on #134, as discussed [here](https://github.com/model-checking/verify-rust-std/pull/134?new_mergebox=true#discussion_r1866459750).

**Comment by tautschnig** [comment]
@MWDZ Now that #134 has been merged, could you please resolve conflicts?

File: library/core/src/num/mod.rs [hunk 1]
```diff
@@ -1831,6 +1831,21 @@ mod verify {
...
+                pub fn $harness_name() {
+                    let num1: $floatType = kani::any::<$floatType>();
+                    let result = unsafe { num1.to_int_unchecked::<$intType>() };
+
+                    assert_eq!(result, num1 as $intType);
+                }
+            )+
+        }
+    }
+/*
```
Comment (celinval) [inline library/core/src/num/mod.rs:1848]: why is this commented out?

**Review by celinval — COMMENTED** [review]
Thanks!

File: library/core/src/num/mod.rs (hunk 1 above)
Comment (Yenyun035) [reply library/core/src/num/mod.rs]: Commented out for only running to_int_unchecked harnesses. Fixed

**Comment by Yenyun035** [comment]
@tautschnig @celinval Thank you. Conflicts resolved. Harnesses un-commented. 👍

**Review by zhassan-aws — APPROVED** [review]

**Review by feliperodri — APPROVED** [review]

**Pull request merged**